django/db/models/sql/query.py

```diff
@@ -1408,13 +1408,19 @@ def build_filtered_relation_q(self, q_object, reuse, branch_negated=False, curre
     def add_filtered_relation(self, filtered_relation, alias):
         filtered_relation.alias = alias
         lookups = dict(get_children_from_q(filtered_relation.condition))
-        for lookup in chain((filtered_relation.relation_name,), lookups):
+        relation_lookup_parts, relation_field_parts, _ = self.solve_lookup_type(filtered_relation.relation_name)
+        if relation_lookup_parts:
+            raise ValueError(
+                "FilteredRelation's relation_name doesn't support lookups "
+                "(got %r)." % filtered_relation.relation_name
+            )
+        for lookup in chain(lookups):
             lookup_parts, field_parts, _ = self.solve_lookup_type(lookup)
             shift = 2 if not lookup_parts else 1
-            if len(field_parts) > (shift + len(lookup_parts)):
+            if len(field_parts) > (shift + len(relation_field_parts)):
                 raise ValueError(
                     "FilteredRelation's condition doesn't support nested "
-                    "relations (got %r)." % lookup
+                    "on clauses deeper than the relation (got %r for %r)." % (lookup, filtered_relation.relation_name)
```

Comment:
Validation based only on a nesting level is not a proper solution, e.g.
```python
Author.objects.annotate(
    book_editor_worked_with=FilteredRelation('book__editor', condition=Q(book__author__name__icontains='book')),
)
``` 
will crash with `django.db.utils.OperationalError: no such column: T4.name`.

Reply:
I'll add a test case specifically for this, thanks

Reply:
Added test to catch this error